quinn-proto/src/connection/assembler.rs

```diff
@@ -205,6 +244,13 @@ impl Chunk {
     }
 }
 
+enum NextState<'a> {
+    Value(PeekMut<'a, Buffer>),
+    Blocked,
+    Retry,
+    Empty,
```

Comment:
Should `Empty` and `Blocked` be combined? I don't see any case where we handle them differently.

Reply:
I'm scrapping this part of the PR to land some of the less controversial parts. I think the `read(max_length) -> Option<Chunk>` API is not expressive enough, and therefore maybe prone to misuse.